Sources/DeepLearning/Tensor.swift

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TensorFlowCore
+
+infix operator .==: ComparisonPrecedence
+
+//===------------------------------------------------------------------------------------------===//
+// Description and Visualization
+//===------------------------------------------------------------------------------------------===//
+
+// String conversion.
+extension Tensor: CustomStringConvertible {
+    /// A textual representation of the tensor.
+    ///
+    /// - Note: use `fullDescription` for a non-pretty-printed description showing all scalars.
+    public var description: String {
+        @_semantics("autodiff.nonvarying")
+        get {
+            return array.description
+        }
+    }
+}
+
+public extension Tensor {
+    /// A textual representation of the tensor. Returns a summarized description if `summarize` is
+    /// true and the element count exceeds twice the `edgeElementCount`.
+    ///
+    /// - Parameters:
+    ///   - lineWidth: The max line width for printing. Used to determine number of scalars to print
+    ///     per line.
+    ///   - edgeElementCount: The maximum number of elements to print before and after summarization
+    ///     via ellipses (`...`).
+    ///   - summarizing: If true, summarize description if element count exceeds twice
+    ///     `edgeElementCount`.
+    @_semantics("autodiff.nonvarying")
```

Comment:
I'd suggest removing this from functions for now and only apply them to common properties.